app/addons.go

```diff
@@ -102,34 +116,124 @@ func (a *App) AddAddonHandler(subject, reply string, request *qms.AddAddonReques
 func (a *App) ListAddonsHandler(subject, reply string, request *qms.NoParamsRequest) {
 	var err error
 
-	log := log.WithField("context", "list addons")
+	ctx, span := qmsinit.InitNoParamsRequest(request, subject)
+	defer span.End()
 
+	log := log.WithField("context", "list addons")
+	sendError := a.sendAddonListResponseError(reply, log)
 	response := qmsinit.NewAddonListResponse()
+	d := db.New(a.db)
 
-	sendError := func(ctx context.Context, response *qms.AddonListResponse, err error) {
+	results, err := d.ListAddons(ctx)
+	if err != nil {
+		sendError(ctx, response, err)
+		return
+	}
+
+	for _, addon := range results {
+		response.Addons = append(response.Addons, addon.ToQMSType())
+	}
+
+	if err = a.client.Respond(ctx, reply, response); err != nil {
+		log.Error(err)
+	}
+}
+
+func (a *App) UpdateAddonHandler(subject, reply string, request *qms.UpdateAddonRequest) {
+	var err error
+
+	log := log.WithField("context", "update addon")
+
+	response := qmsinit.NewAddonResponse()
+
+	sendError := func(ctx context.Context, response *qms.AddonResponse, err error) {
 		log.Error(err)
 		response.Error = serrors.NatsError(ctx, err)
 		if err = a.client.Respond(ctx, reply, response); err != nil {
 			log.Error(err)
 		}
 	}
```

Comment:
looks like this is the same function again too, though I think in this case it's completely identical to `sendAddonResponseError`'s return value.